state/txindex/kv/kv.go

```diff
@@ -417,6 +436,13 @@ func (txi *TxIndex) match(
 			if strings.Contains(extractValueFromKey(it.Key()), c.Operand.(string)) {
 				tmpHashes[string(it.Value())] = it.Value()
 			}
+
+			// Potentially exit early.
+			select {
```

Comment:
Shouldn't we need to add this escape feature to for loop in `c.Op == query.OpEqual`and in below intersection? If not, it will keep loading all tx index to memory and doing unnecessary intersection work.

https://github.com/tendermint/tendermint/blob/0c7827a226cc0928d4903ea5db60db8a3c5787a1/state/txindex/kv/kv.go#L418

https://github.com/tendermint/tendermint/blob/0c7827a226cc0928d4903ea5db60db8a3c5787a1/state/txindex/kv/kv.go#L464